### PR TITLE
#5544: Add output tensors parameter to moreh_nll_loss op

### PR DESCRIPTION
**Pull request by dongjin-na**
I added output tensor parameters to these OPs below refer to issue https://github.com/tenstorrent-metal/tt-metal/issues/5044 .

- moreh_nll_loss
- moreh_nll_loss_backward

Passed the relevant test below.

- tests/tt_eager/python_api_testing/unit_testing/misc/test_moreh_nll_loss.py

### CODE REVIEW

**Review by eyonland — COMMENTED** [review]
Consider using operation::ProgramWithOptionalOutputTensors as mentioned here:
https://tenstorrent-metal.github.io/tt-metal/latest/ttnn/ttnn/dependencies/tt_lib.html#new-device-operation-with-optional-output-tensors

**Review by eyonland — APPROVED** [review]
Consider using operation::ProgramWithOptionalOutputTensors as mentioned here:
https://tenstorrent-metal.github.io/tt-metal/latest/ttnn/ttnn/dependencies/tt_lib.html#new-device-operation-with-optional-output-tensors

**Comment by dongjin-na** [comment]
@mywoodstock , @TT-BrianLiu , @tt-aho
dear code owners, would you please review this PR? 
This PR is the same as https://github.com/tenstorrent/tt-metal/pull/7101

File: tests/tt_eager/python_api_testing/unit_testing/misc/test_moreh_nll_loss.py [hunk 1]
```diff
@@ -140,7 +140,9 @@ def get_tt_tensors_2d(torch_input, torch_target, torch_weight, torch_divisor, to
 )
 @pytest.mark.parametrize("ignore_index", [0, -1])
 @pytest.mark.parametrize("reduction_mean", [True, False])
-def test_moreh_nll_loss_4d(shape, ignore_index, reduction_mean, device):
+@pytest.mark.parametrize("has_output", [True, False])
+def test_moreh_nll_loss_4d(shape, ignore_index, reduction_mean, has_output, device):
+    # def test_moreh_nll_loss_4d(shape, ignore_index, reduction_mean, device):
     device.enable_program_cache()
```
Comment (TT-BrianLiu) [inline tests/tt_eager/python_api_testing/unit_testing/misc/test_moreh_nll_loss.py:146]: Use program cache fixture?

**Review by TT-BrianLiu — APPROVED** [review]
Minor comment. Looks good to merge.

File: tests/tt_eager/python_api_testing/unit_testing/misc/test_moreh_nll_loss.py (hunk 1 above)
Comment (dongjin-na) [reply tests/tt_eager/python_api_testing/unit_testing/misc/test_moreh_nll_loss.py]: Applied it.

**Pull request merged**